### ISSUE TITLE
Fleau silently fails when you try to use an unexistant parser (and the underlying exception is misleading)

### ISSUE THREAD

**Issue by jankeromnes** [espadrine/fleau#4]
When you try to use an unexistant parser in one of your fleau templates (e.g. you use https://github.com/espadrine/sc, but accidentally compile your templates with `Camp.template(paths)` **before** your custom parsers are added with `Camp.templateReader.parsers.customParser = x => {}`), fleau fails silently (no error or exception or anything visible). I believe any exceptions are silently catched by some weird promise thing.

However, if you surround the call to the actual failing template with a try/catch, you discover a very confusing exception:

```
TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at evalmachine.<anonymous>:65:134
    at ContextifyScript.Script.runInContext (vm.js:32:29)
    at ContextifyScript.Script.runInNewContext (vm.js:38:15)
    at /home/ubuntu/workspace/node_modules/fleau/fleau.js:247:12
    at /home/ubuntu/workspace/node_modules/camp/lib/camp.js:88:14
    at Array.map (native)
    at ServerResponse.res.template (/home/ubuntu/workspace/node_modules/camp/lib/camp.js:82:37)
    at Object.exports.settingsPage (/home/ubuntu/workspace/lib/routes.js:258:12)
    at app.route (/home/ubuntu/workspace/app.js:274:12)
```

This is not helpful at all, and you have to guess by yourself that you're using an unexistant parser by trial and error.

Also, you discover that your `scope` objects are being permanently polluted with many `$_` prefixed attributes (it would be nice if fleau's runner function copied its `scope` argument, e.g. with `var scope = JSON.parse(JSON.stringify(scope))`, before altering it).

In summary, it would be nice to:
- [ ] Not silently catch all exceptions
- [ ] Make it more obvious that you're trying to use a parser that doesn't exist
- [ ] Make fleau's runner function duplicate its `scope` object before adding its `$_` prefixed attributes to it.

**Comment by espadrine** [comment]
Is this a fleau 16.2.0 bug? I cannot reproduce it.

**Comment by jankeromnes** [comment]
It's actually `fleau`'s dependency module `promise` that silently catches all fleau errors.

I'll file the bug on fleau instead.

**Comment by jankeromnes** [comment]
- The "Not silently catch all exceptions" bug is now https://github.com/espadrine/fsos/issues/3
- The "Make it more obvious that you're trying to use a parser that doesn't exist" bug is invalid (when the error is logged, it's pretty obvious that the parser doesn't exist)
- The "Make fleau's runner function duplicate its `scope` object before adding its `$_` prefixed attributes to it" should probably still be addressed.

**Comment by jankeromnes** [comment]
For the silenced errors, see also https://github.com/then/promise/issues/137.
